### ISSUE TITLE
market fsm/storage deal status should support SnapDeal

### ISSUE THREAD

**Issue by jennijuju** [filecoin-project/go-fil-markets#662]
Fip 0019 introduces SnapDeal that allows SP to put deal into cc sectors by doing `ProveReplicaUpdate` without  reseal the sector.

see the sector fsm here https://github.com/filecoin-project/lotus/blob/d645c5fbabe2c087c21058d7a5eeda09da7a0f0a/extern/storage-sealing/fsm.go#L137-L174

and storage deal status should reflect that https://github.com/filecoin-project/go-fil-markets/blob/0f905124261c3df7a9c7691faa320657af8ea6e2/storagemarket/dealstatus.go#L191-L192

**Comment by jacobheun** [comment]
@jennijuju can you clarify what outcome is needed here? Is this for storage deals list? Not quite sure what end state we're hoping to expose here.

**Comment by jennijuju** [comment]
yes. ```...opmpcyjq  2890    StorageDealAwaitingPreCommit  t12j6glkr4vetge3iqw5vxwl4zubjsxlgt4c5iiji  2MiB   0 FIL  603647```

for SnapDeals, there is no precommit or prove commit(sealing), but only a `ProveReplicaUpdate` step - this may take between 16min to an hour based on the hardware.

I would suggest to add a state called `StorageDealSnapping`
